src/Consumer/KafkaConsumerBuilder.php

```diff
@@ -359,4 +364,17 @@ private function registerCallbacks(KafkaConfiguration $conf): void
             $conf->setOffsetCommitCb($this->rebalanceCallback);
         }
     }
+
+    /**
+     * @param array $partitions
+     * @param int $offset
+     */
+    private function validateSubscription(array $partitions, int $offset): void
+    {
+        if (0 <= $offset && [] === $partitions) {
```

Comment:
shouldn't we include the case of multiple partitions as well?
`if (0 <= $offset && 1 !== count($partitions))`